### ISSUE TITLE
UI Editor: Editor crashes when opening the Asset Picker in the Sprite path field if search suggestions are displayed

### ISSUE THREAD

**Issue by LB-BartoszDudzinski** [o3de/o3de#4190]
**Describe the bug**
Editor crashes when pressing the Asset Picker button, if search suggestions pointing to the same asset that was already assigned are displayed.

**To Reproduce**
1. Open the UI Editor.
2. Create a new element.
3. Add an Image component to it.
4. Assign any image to the Sprite path.
5. Click the text field and type the name of the used asset.
6. While the search suggestions are still displayed, press the Asset Picker button (the folder icon on the right).

**Expected behavior**
The Asset Picker is opened. The Editor remains stable.

**Call Stack Trace:**
```
25) 00007FF8986BC347 (KERNELBASE) : UnhandledExceptionFilter
24) 00007FF89AB151B0 (ntdll) : memset
23) 00007FF89AAFC766 (ntdll) : _C_specific_handler
22) 00007FF89AB120CF (ntdll) : _chkstk
21) 00007FF89AAC1454 (ntdll) : RtlRaiseException
20) 00007FF89AB10BFE (ntdll) : KiUserExceptionDispatcher
19) 00007FFFF2CA58C6 (EditorLib) : AzToolsFramework::AssetBrowser::AssetPickerDialog::`scalar deleting destructor'
18) 00007FF84B98551B (Qt5Core) : QObjectPrivate::deleteChildren
17) 00007FF84C412060 (Qt5Widgets) : QWidget::~QWidget
16) 00007FF84CDB107E (AzQtComponents) : AzQtComponents::WindowDecorationWrapper::`vector deleting destructor'
15) 00007FF84B98551B (Qt5Core) : QObjectPrivate::deleteChildren
14) 00007FF84C412060 (Qt5Widgets) : QWidget::~QWidget
13) 00007FFFF8BA2F85 (LyShine.Editor) : PropertySpriteCtrl::`scalar deleting destructor'
12) C:\O3DE\o3de\Code\Framework\AzToolsFramework\AzToolsFramework\UI\PropertyEditor\PropertyRowWidget.cpp (154) : AzToolsFramework::PropertyRowWidget::Clear
11) C:\O3DE\o3de\Code\Framework\AzToolsFramework\AzToolsFramework\UI\PropertyEditor\ReflectedPropertyEditor.cpp (1144) : AzToolsFramework::ReflectedPropertyEditor::Impl::ReturnAllToPool
10) C:\O3DE\o3de\Code\Framework\AzToolsFramework\AzToolsFramework\UI\PropertyEditor\ReflectedPropertyEditor.cpp (886) : AzToolsFramework::ReflectedPropertyEditor::InvalidateAll
 9) C:\O3DE\o3de\Code\Framework\AzToolsFramework\AzToolsFramework\UI\PropertyEditor\ReflectedPropertyEditor.cpp (2317) : AzToolsFramework::ReflectedPropertyEditor::DoRefresh
 8) 00007FF84B988631 (Qt5Core) : QObject::event
 7) 00007FF84C4187C5 (Qt5Widgets) : QWidget::event
 6) 00007FF84C4BA090 (Qt5Widgets) : QFrame::event
 5) 00007FF84C3F49B0 (Qt5Widgets) : QApplicationPrivate::notify_helper
 4) 00007FF84C3F3A33 (Qt5Widgets) : QApplication::notify
 3) 00007FF84B961B4A (Qt5Core) : QCoreApplication::notifyInternal2
 2) 00007FF84B9638C5 (Qt5Core) : QCoreApplicationPrivate::sendPostedEvents
 1) 00007FF87DAA383F (qwindows) : qt_plugin_query_metadata
```
**Screenshots**

https://user-images.githubusercontent.com/86952009/133766367-0ac7313d-0311-4ed0-8237-b630c6853577.mp4

**Desktop/Device:**
- Device: PC
- OS: Windows
- Version 10
- CPU Intel I5-9600k
- GPU RTX 2060
- Memory 16GB

**Additional context**
[Editor.log](https://github.com/o3de/o3de/files/7184814/Editor.log)
[error.log](https://github.com/o3de/o3de/files/7184815/error.log)
[dump.zip](https://github.com/o3de/o3de/files/7184816/dump.zip)

**Comment by AMZN-AlexOteiza** [comment]
Fixed in https://github.com/o3de/o3de/pull/5495

**Comment by LB-BartoszDudzinski** [comment]
Verified as fixed in Development (3791149) however, the issue still occurs in Stabilization_2110 (7031147) branch.
